be/src/io/cache/block_file_cache.cpp

```diff
@@ -1203,39 +1313,48 @@ bool BlockFileCache::try_reserve_from_other_queue_by_hot_interval(
                 DCHECK(file_block->_download_state == FileBlock::State::DOWNLOADED);
                 to_evict.push_back(cell);
                 removed_size += cell_size;
+                remove_size_per_type += cell_size;
             }
         }
+        *(_evict_by_heat_metrics_matrix[cache_type][cur_type]) << remove_size_per_type;
     }
     remove_file_blocks(to_evict, cache_lock);
 
     return !is_overflow(removed_size, size, cur_cache_size);
 }
 
-bool BlockFileCache::is_overflow(size_t removed_size, size_t need_size, size_t cur_cache_size,
-                                 bool is_ttl) const {
+bool BlockFileCache::is_overflow(size_t removed_size, size_t need_size,
+                                 size_t cur_cache_size) const {
     bool ret = false;
     if (_disk_resource_limit_mode) {
         ret = (removed_size < need_size);
     } else {
         ret = (cur_cache_size + need_size - removed_size > _capacity);
     }
-    if (is_ttl) {
-        size_t ttl_threshold = config::max_ttl_cache_ratio * _capacity / 100;
-        return (ret || ((cur_cache_size + need_size - removed_size) > ttl_threshold));
-    }
     return ret;
 }
 
 bool BlockFileCache::try_reserve_from_other_queue_by_size(
-        std::vector<FileCacheType> other_cache_types, size_t size,
+        FileCacheType cur_type, std::vector<FileCacheType> other_cache_types, size_t size,
         std::lock_guard<std::mutex>& cache_lock) {
     size_t removed_size = 0;
     size_t cur_cache_size = _cur_cache_size;
     std::vector<FileBlockCell*> to_evict;
+    // we follow the privilege defined in get_other_cache_types to evict
```

Comment:
try to rename `get_other_cache_types` with more meaningful name e.g. `get_other_cache_types_to_evict`